### PR TITLE
Task/break apart forall segment test instantiations

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR is a proposal for breaking apart tests into different executables to make it clearer what's going on and keep the contents of the files cleaner.
- Mostly file, type alias, and test suite renaming.

### CODE REVIEW

File: test/functional/forall/tests/test-forall-utils.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#ifndef __TEST_FORALL_UTILS_HPP__
-#define __TEST_FORALL_UTILS_HPP__
+#ifndef __TEST_FORALL_SEGMENT_UTILS_HPP__
+#define __TEST_FORALL_SEGMENT_UTILS_HPP__
```
Comment (mdavis36) [inline test/functional/forall/tests/test-forall-utils.hpp]: I had originally desired the utils header to be dependant of the test group/type, did you want to make individual utils for each group or is this a typo?

File: test/functional/forall/tests/test-forall-utils.hpp [hunk 2]
```diff
@@ -63,4 +83,4 @@ void deallocateForallTestData(Resource& work_res,
   host_res.deallocate(test_array);
 }
 
-#endif  // __TEST_FORALL_UTILS_HPP__
+#endif  // __TEST_FORALL_SEGMENT_UTILS_HPP__
```
Comment (mdavis36) [inline test/functional/forall/tests/test-forall-utils.hpp]: Same here.

File: test/functional/forall/tests/test-forall-rangesegment.hpp [hunk 3]
```diff
@@ -59,20 +59,20 @@ template <typename INDEX_TYPE, typename WORKING_RES, typename EXEC_POLICY,
...
 {
-  ForallRangeSegmentFunctionalTest<INDEX_TYPE, WORKING_RES, EXEC_POLICY>(-5, 0);
-  ForallRangeSegmentFunctionalTest<INDEX_TYPE, WORKING_RES, EXEC_POLICY>(-5, 5);
+  ForallRangeSegmentTest<INDEX_TYPE, WORKING_RES, EXEC_POLICY>(-5, 0);
+  ForallRangeSegmentTest<INDEX_TYPE, WORKING_RES, EXEC_POLICY>(-5, 5);
 }
 
 
-TYPED_TEST_P(ForallFunctionalSegmentTest, RangeSegmentForall)
+TYPED_TEST_P(ForallSegmentTest, RangeSegmentForall)
```
Comment (mdavis36) [inline test/functional/forall/tests/test-forall-rangesegment.hpp]: Do we want to make functional tests have the same name format of <Test>FunctionalTest to reflect the same naming standard of the unit tests?

File: test/functional/forall/tests/test-forall-utils.hpp (hunk 1 above)
Comment (rhornung67) [reply test/functional/forall/tests/test-forall-utils.hpp]: That was a mistake. It's changed back to what it was.

File: test/functional/forall/tests/test-forall-utils.hpp (hunk 2 above)
Comment (rhornung67) [reply test/functional/forall/tests/test-forall-utils.hpp]: Changed back.

File: test/functional/forall/tests/test-forall-rangesegment.hpp (hunk 3 above)
Comment (rhornung67) [reply test/functional/forall/tests/test-forall-rangesegment.hpp]: I thought about that. To me adding 'Functional' makes things too cumbersome. If it's not a unit test, then it's a functional test in my mind. But, I can change it back if you think it's important to keep it.

Comment (mdavis36) [reply test/functional/forall/tests/test-forall-rangesegment.hpp]: No I think that's reasonable, just wanted to check

**Pull request merged**